src/components/layout/navbar.tsx

```diff
@@ -1,21 +1,46 @@
 import React from 'react';
+import { useHistory } from 'react-router-dom';
 import { useDispatch, useSelector } from 'react-redux';
-
 import { LanguagePicker } from '~/components/editor/language-picker';
 import { getIsLoggedIn } from '~/store/getters/auth';
 import { logoutUser } from '~/store/action/auth';
 import { setActiveModal } from '~/store/action/ui';
 import { LOGIN_MODAL } from '~/constants/modal';
+import { getCurrentSource, getStdin, getSelectedLanguage } from '~/store/getters/editor';
+import { clearJwt } from '~/utils/jwt';
+import { saveUpdateCode } from '~/tasks/save-update-code';
 
 export const Navbar: React.FC = () => {
   const dispatch = useDispatch();
   const toggleOverlay = () => dispatch(setActiveModal(LOGIN_MODAL));
   const isLoggedIn = useSelector(getIsLoggedIn());
+  const data = {
+    lang: JSON.stringify(useSelector(getSelectedLanguage())),
+    source: useSelector(getCurrentSource()),
+    input: useSelector(getStdin()),
+  };
+
+  const history = useHistory();
 
   const logout = () => {
     dispatch(logoutUser());
+    clearJwt();
   };
 
+  async function saveCode() {
```

Comment:
can we do task here as well? Just for the sake of having a running state and maybe we can show Saving on the button as well using that.

Reply:
Fixed.